src/factories/EmployeeFactory.java

```diff
@@ -0,0 +1,33 @@
+package factories;
+
+import mappers.EmployeeMapper;
+import models.EmployeeModel;
+import services.EmployeeService;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class EmployeeFactory {
+
+    private EmployeeMapper employeeMapper;
+    private EmployeeService employeeService;
+
+    public EmployeeFactory() {
+        this.employeeMapper = new EmployeeMapper();
+        this.employeeService = new EmployeeService();
+    }
+
+    public HashMap<String, EmployeeModel> build(ArrayList<String> fetchedData) {
+
+        HashMap<String, EmployeeModel> mappedEmployees = employeeMapper.map(fetchedData);
+
+
+        for (EmployeeModel employeeModel : mappedEmployees.values()) {
+
+            String employee = employeeModel.getLogInCode();
+            HashMap<String, EmployeeModel> employeeLogIn = employeeService.get(employee);
+            employeeModel.setEmployeeList(employeeLogIn);
```

Comment:
❌ This factory is basically a repeat of the mapper code.   The mapper code looks good tho.  I would suggest that you make 1 of 2 changes.  

Either, you throw away this factory because it is basically just a rehash of employeeMapper (basically meaning that nothing extra is needed to compose a employee model other than running the mapper) 
If you remove this factory, line 21 will need to be copied and moved from here out to the service that called this factory.  Line 21 should be all you need in the service in order to get a hash map of employees.


Or you can make the factory understand how to iterate over the fetchedData (for loop over the fetchedData here in this factory) and then for each of the lines in the fetchedData you would call to the employee mapper to transform a single line into an employee object.  If this is done, you will need to clean up the mapper to remove the for loop.  Basically your mapper will take in ONE line of fetchedData and map to ONE employee object and then your factory will collect all the built up employee objects into a hash map.